src/Data/List.purs

```diff
@@ -257,6 +253,18 @@ uncons :: forall a. List a -> Maybe { head :: a, tail :: List a }
 uncons Nil = Nothing
 uncons (x : xs) = Just { head: x, tail: xs }
 
+-- | Break a list into its last element, and the preceding elements,
+-- | or `Nothing` if the list is empty.
+-- |
+-- | Running time: `O(n)`
+unsnoc :: forall a. List a -> Maybe { init :: List a, last :: a }
+unsnoc lst = unsnocHelper lst Nil <#> \h -> { init: reverse h.revInit, last: h.last }
+
+unsnocHelper :: forall a. List a -> List a -> Maybe { revInit :: List a, last :: a }
```

Comment:
I don't know the extent to which JS engines optimize under functions, but I'd say it's better to be consistent at this point.